processing/data_collection/gazette/spiders/sp_guaruja.py

```diff
@@ -8,30 +10,105 @@
+        calendar_id = calendar_id[16:]
+
+        target_date = self.start_date
+        data = {
+            "action": "mec_full_calendar_switch_skin",
+            "skin": "monthly",
+            "atts[skin]": "full_calendar",
+            "atts[show_past_events]": "1",
+            "atts[id]": calendar_id,
+            "apply_sf_date": "1",
+        }
+
+        while target_date <= self.end_date:
+            data.update(
+                {
+                    "sf[month]": f"{target_date.month:02}",
+                    "sf[year]": str(target_date.year),
+                }
+            )
+            yield scrapy.FormRequest(
+                url="https://www.guaruja.sp.gov.br/wp-admin/admin-ajax.php",
+                formdata=data,
+                method="POST",
+            )
+            target_date = target_date + relativedelta(months=1)
 
     def parse(self, response):
-        months = response.css("div.span12 a::attr(href)").extract()
-        for month_url in months:
-            yield scrapy.Request(month_url, self.parse_items)
-
-    def parse_items(self, response):
-        gazettes = response.css("div.span12 p")
-        for gazette in gazettes:
-            date = gazette.css("a ::text").extract_first()
-            is_extra_edition = "parte" in date and "parte1" not in date
-            extra_editions = [is_extra_edition, True]
-            date = parse(date, languages=["pt"]).date()
-            urls = gazette.css("a::attr(href)").extract()
-            for url, is_extra_edition in zip(urls, extra_editions):
-                yield Gazette(
-                    date=date,
-                    file_urls=[url],
-                    is_extra_edition=is_extra_edition,
-                    territory_id=self.TERRITORY_ID,
-                    power="executive_legislature",
-                    scraped_at=datetime.utcnow(),
-                )
+
+        clean_response = self.clean_response_body(response)
+        for event_link in clean_response.xpath('//h4[@class="mec-event-title"]'):
+            file_url = event_link.xpath("a/@href").get()
+            event_date = event_link.xpath(
+                'ancestor::div[@class="mec-calendar-events-sec"]/@data-mec-cell'
+            ).get()
+            event_date = dt.datetime.strptime(event_date, "%Y%m%d").date()
+            if event_date > self.end_date:
+                # We could break here, but god only
+                # knows if they sort the output.
+                continue
+
+            yield Gazette(
+                date=event_date,
+                file_urls=[file_url],
+                is_extra_edition=False,
+                territory_id=self.TERRITORY_ID,
+                power="executive_legislature",
+                scraped_at=dt.datetime.utcnow(),
+            )
+
+    @classmethod
+    def clean_response_body(cls, response):
+        # Response body here is a messed up double back-slashed string
```

Comment:
I'm curious about this one here. Isn't it the JavaScript the only one affected by this issue? 🤔

Reply:
Not sure I understood what you meant there. I guess the javascript parses this somehow. If you know a better way of cleaning this string, please let me know.